object-store/src/main/java/org/hypertrace/config/objectstore/ConfigObject.java

```diff
@@ -1,9 +1,10 @@
 package org.hypertrace.config.objectstore;
 
 import java.time.Instant;
+import java.util.Optional;
 
 public interface ConfigObject<T> {
-  T getData();
+  Optional<T> getData();
```

Comment:
I think there was a misunderstanding here - getData should not be made optional. _Delete's_ return can be made optional. In every scenario beside delete, we don't want this to be empty, so now we've just made dozens of usages in this PR alone have to call `.get()` on an optional, which is a big code smell. If we have to call `.get()`, we've messed up.

My suggestion had been to change the object stores so instead of
```
public Optional<ContextualConfigObject<T>> deleteObject(
```

return something like
```
public DeleteResult<T> deleteObject(
```
Or potentially (this one probably will make it cleaner, since most result handling just works with the optional today)
```
public Optional<DeletedContextualConfigObject<T>> deleteObject(
```

In this way, we're only breaking consumers of the delete API and in the latter one, only that subset of consumers that then needs to actually read data from the result.

Reply:
Right, got it. Updated it now.